reference/src/layout/integers-floatingpoint.md

```diff
@@ -37,6 +37,25 @@ Note: on all platforms that Rust's currently supports, the size and alignment of
 
 For full ABI compatibility details, see [Gankro’s post](https://gankro.github.io/blah/rust-layouts-and-abis/#the-layoutsabis-of-builtins).
 
+## Fixed-width integer types
+
+Rust's fixed-width integer types `{i,u}{8,16,32,64}` have the same layout as the
+C fixed-width integer types from the `<stdint.h>` header
+`{u,}int{8,16,32,64}_t`. That is:
```

Comment:
This section overlaps in scope with "Relationship to C integer hierarchy". Shouldn't the two be merged?

Reply:
That section states some of the properties of C native integer types (e.g. `int`) and states that Rust types are not guaranteed to be layout compatible with those.

I could move it to a note / sub-section of this one (the fixed-integer one). I think it should not only mention what C guarantees about its native integer types (e.g. `int`), but also what it does not guarantee (e.g. padding bits, representation of negative numbers, etc.), to explain why Rust fixed-width integer types aren't guaranteed to always match to any of those. 

It should mention that in practice one can, however, use "implementation-defined" information to properly interface with these integer types by using the appropriate Rust types (and this is in fact what `libc` does, so we want to explicitly mention that this is ok). 

We should also note that in practice, all platforms that Rust currently support have integers with 2s complement representation, without padding bits, etc.